server/src/main/java/org/elasticsearch/ingest/IngestService.java

```diff
@@ -1191,19 +1191,8 @@ private static Optional<Pipelines> resolvePipelinesFromMetadata(
             return Optional.empty();
         }
 
-        String defaultPipeline = null;
-        String finalPipeline = null;
-        final Settings indexSettings = indexMetadata.getSettings();
-        if (IndexSettings.DEFAULT_PIPELINE.exists(indexSettings)) {
-            // find the default pipeline if one is defined from an existing index setting
-            defaultPipeline = IndexSettings.DEFAULT_PIPELINE.get(indexSettings);
-        }
-        if (IndexSettings.FINAL_PIPELINE.exists(indexSettings)) {
-            // find the final pipeline if one is defined from an existing index setting
-            finalPipeline = IndexSettings.FINAL_PIPELINE.get(indexSettings);
-        }
-
-        return Optional.of(new Pipelines(defaultPipeline, finalPipeline));
+        final Settings settings = indexMetadata.getSettings();
+        return Optional.of(new Pipelines(IndexSettings.DEFAULT_PIPELINE.get(settings), IndexSettings.FINAL_PIPELINE.get(settings)));
```

Comment:
oh nice! this method returns the default_value (`_none`) in case the settings does not exist. cool